vm-validator/src/unit_tests/vm_validator_test.rs

```diff
@@ -33,28 +31,25 @@ impl TestValidator {
         let storage = start_storage_service(&config);
 
         // setup execution
-        let client_env = Arc::new(EnvBuilder::new().build());
         let storage_read_client: Arc<dyn StorageRead> = Arc::new(StorageReadServiceClient::new(
-            Arc::clone(&client_env),
             &config.storage.address,
             config.storage.port,
         ));
+
         let storage_write_client = Arc::new(StorageWriteServiceClient::new(
-            Arc::clone(&client_env),
             &config.storage.address,
             config.storage.port,
-            None,
         ));
 
         // Create executor to initialize genesis state. Otherwise gprc will report error when
         // fetching data from storage.
-        let _executor = Executor::<LibraVM>::new(
-            Arc::clone(&storage_read_client) as Arc<dyn StorageRead>,
-            storage_write_client,
-            config,
-        );
+        let _executor = Executor::<LibraVM>::new(storage_read_client, storage_write_client, config);
 
-        let vm_validator = VMValidator::new(config, storage_read_client, rt.handle().clone());
+        let read_client: Arc<dyn StorageRead> = Arc::new(StorageReadServiceClient::new(
```

Comment:
Please leave a comment here that you can't share clients, otherwise it seems strange to create two of these.